addons/account_tax_exigible/__init__.py

```diff
@@ -0,0 +1,43 @@
+import models
+from openerp import SUPERUSER_ID
+
+def _migration_scrip(cr, registry):
+    move_ids = registry['account.move'].search(cr, SUPERUSER_ID, [])
+    moves = registry['account.move'].browse(cr, SUPERUSER_ID, move_ids)
+    for move in moves:
+        for line in move.line_ids:
+            taxes = False
+            if line.tax_line_id:
+                taxes = [{'use_cash_basis': registry['account.tax'].browse(vals['tax_line_id']).use_cash_basis}]
+            if line.tax_ids:
+                taxes = registry['account.move.line'].resolve_2many_commands('tax_ids', line.tax_ids.ids])
+        if taxes and any([tax['use_cash_basis'] for tax in taxes]) and not line.tax_exigible:
+            line.tax_exigible = False
+        else:
+            move.line_ids.write({'tax_exigible': True})
+            break
+    reconcile_ids = registry['account.partial.reconcile'].search(cr, SUPERUSER_ID, [])
+    reconciles = registry['account.partial.reconcile'].browse(cr, SUPERUSER_ID, reconcile_ids)
+    for reconcile in reconciles:
+        for move in (reconcile.debit_move_id.move_id, reconcile.credit_move_id.move_id):
+            for line in move.line_ids:
+                if not line.tax_exigible:
+                    for tax in line.tax_ids:
+                        if tax.use_cash_basis:
+                            registry['account.move.line']create({
+                                'name': '/',
+                                'debit': line.debit_cash_basis - line.debit * move.matched_percentage,
+                                'credit': line.credit_cash_basis - line.credit * move.matched_percentage,
+                                'account_id': line.account_id.id,
+                                'tax_ids': [(6, 0, [tax.id])],
+                                'tax_exigible': True,
+                                'move_id': move.id,
+                            })
+                            registry['account.move.line']create({
+                                'name': '/',
+                                'credit': line.debit_cash_basis - line.debit * move.matched_percentage,
+                                'debit': line.credit_cash_basis - line.credit * move.matched_percentage,
+                                'account_id': line.account_id.id,
+                                'move_id': move.id,
+                                })
+    return
```

Comment:
you can choose the existing line of the move that has the same account_id as the account_id or refund_id set on the tax